reconcile/gitlab_fork_compliance.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from gitlab import MAINTAINER_ACCESS
+
+from reconcile import queries
+from utils.gitlab_api import GitLabApi
+from utils.gitlab_api import MRState
+
+
+_LOG = logging.getLogger(__name__)
+
+QONTRACT_INTEGRATION = 'gitlab-fork-compliance'
+
+MAINTAINERS_GROUP = 'app-sre'
+BOT_USER = 'devtools-bot'
+BLOCKED_LABEL = 'blocked/devtools-bot-access'
+MSG = ('@{user}, this fork of {project_name} is not shared with '
+       '[{bot}](/{bot}) as "Maintainer". !!Please '
+       '[add the user to the project]({source_project_url}/project_members) '
+       'and retest by commenting "[test]" on the merge request.')
+
+
+def run(project_id, dry_run=False):
+    instance = queries.get_gitlab_instance()
+    settings = queries.get_app_interface_settings()
+
+    gitlab_cli = GitLabApi(instance, project_id=project_id,
+                           settings=settings)
+
+    group = gitlab_cli.gl.groups.get(MAINTAINERS_GROUP)
+    maintainers = group.members.list()
+    bot = gitlab_cli.get_user(BOT_USER)
+
+    for mr in gitlab_cli.get_merge_requests(state=MRState.OPENED):
```

Comment:
unlike gitlab-owners, this integration should run as part of a pr check of every MR.
so it should get the merge request id as an argument and only act on that MR.